src/master-playlist-controller.js

```diff
@@ -711,24 +711,38 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return false;
     }
 
-    // when the video is a live stream
-    if (!media.endList) {
+    // when the video is a live stream, or has a specific start time:
+    if (!media.endList || media.start) {
       const seekable = this.seekable();
 
       if (!seekable.length) {
-        // without a seekable range, the player cannot seek to begin buffering at the live
-        // point
+        // without a seekable range, the player cannot seek to a specific time
         return false;
       }
 
+      const seekableTime = seekable.end(0);
+      let startTime = seekableTime;
+
+      if (media.start) {
+        const offset = media.start.timeOffset;
```

Comment:
It might be clearer (and easier to test independently) to wrap all of this inside of another function. e.g., `startTime = mediaStartTime(media.start.timeOffset, media.start.precise, seekable.end(0))`